stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/compression/UnzipBytes.java

```diff
@@ -0,0 +1,175 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ * Native function ballerina.compression:unzipBytes.
+ *
+ * @since 0.961
+ */
+@BallerinaFunction(
+        packageName = "ballerina.compression",
+        functionName = "unzipBytes",
+        args = {@Argument(name = "content", type = TypeKind.BLOB),
+                @Argument(name = "destDir", type = TypeKind.STRING)},
+        isPublic = true
+)
+public class UnzipBytes extends AbstractNativeFunction {
+
+    private static final Logger log = LoggerFactory.getLogger(UnzipBytes.class);
+
+    /**
+     * File content as byte array defined.
+     */
+    private static final int SRC_AS_BYTEARRAY_FIELD_INDEX = 0;
+
+    /**
+     * File path of the destination directory.
+     */
+    private static final int DEST_PATH_FIELD_INDEX = 0;
+
+    /**
+     * Decompress/unzip byte arrays/blob.
+     *
+     * @param fileContentAsByteArray file content as a byte arry
+     * @param outputFolder           destination folder
+     */
+    protected static void decompress(byte[] fileContentAsByteArray, String outputFolder) {
+        ZipInputStream zin = null;
+        try {
+            File outdir = new File(outputFolder);
+            zin = new ZipInputStream(new ByteArrayInputStream(fileContentAsByteArray));
+            ZipEntry entry;
+            String name, dir;
+            while ((entry = zin.getNextEntry()) != null) {
+                name = entry.getName();
+                if (entry.isDirectory()) {
+                    mkdirs(outdir, name);
+                    continue;
+                }
+        /* this part is necessary because file entry can come before
+         * directory entry where is file located
+         */
+                dir = dirpart(name);
+                if (dir != null) {
+                    mkdirs(outdir, dir);
+                }
+
+                extractFile(zin, outdir, name);
+            }
+        } catch (IOException e) {
+            log.debug("I/O Exception when processing files ", e);
+            log.error("I/O Exception when processing files " + e.getMessage());
+        } finally {
+            try {
+                if (zin != null) {
+                    zin.close();
+                }
+            } catch (IOException e) {
+                log.debug("I/O Exception when closing the input stream ", e);
+                log.error("I/O Exception when closing the input stream " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Extract files from the zipInputStream.
+     *
+     * @param in     zipInputStream object
+     * @param outdir output directory file
+     * @param name
+     * @throws IOException
```

Comment:
This method seems not throwing any IOException

Reply:
Fixed with f97b66c